enhancements/security/selinux-operator.md

```diff
@@ -0,0 +1,269 @@
+    (allow process self (tun_socket (relabelfrom)))
+    (allow process self (tun_socket (relabelto)))
+    (allow process self (tun_socket (attach_queue)))
+    (allow process sysfs_t (file (write)))
+    (allow process tmp_t (dir (write add_name open getattr setattr read link search remove_name reparent lock ioctl)))
+    (allow process tmp_t (file (setattr open read write create getattr append ioctl lock)))
+    (allow process container_share_t (dir (write add_name)))
+    (allow process container_share_t (file (create setattr write )))
+    (allow process container_var_run_t (dir (write add_name open getattr setattr read link search remove_name reparent lock ioctl)))
+    (allow process container_var_run_t (file (setattr open read write create getattr append ioctl lock)))
+status:
+  usage: virt-launcher_my-namespace.process
+  state: INSTALLED
+```
+
+Where:
+
+* `spec.apply`: Is a flag that allows us to tell the operator if the policy is to be
+  applied (installed) or not. Having this flag allows policy writers and system
+  administrators to verify the policy before taking it into use in the system.
+  Automated tools such as the aforementioned `selinux-policy-helper-operator`
+  would generate policies with this flag set to `false`.
+* `spec.policy`: Contains the actual lines of policy in
+  [CIL language](https://github.com/SELinuxProject/cil/wiki). The decision for
+  this language is because `Udica` already writes policies in such a language,
+  and it's an easier to read format than the typical SELinux type enforcement
+  (te) language.
+* `status.usage`: This string indicates how the user is supposed to use this
+  policy in their `Pod` templates. The format is as follows:
+  `<policy name>_<policy namespace>.process>`
+* `status.state`: This represents the current status of the policy in the
+  cluster. The accepted states are: `PENDING`, `IN-PROGRESS`, `INSTALLED`,
+  `ERROR`. A typical reason why a policy would end up in an ERROR state would
+  be that it has errors in the syntax.
+
+
+### Implementation Details/Notes/Constraints
+
+* Installing policies may take some seconds for each of the clusters, so pods
+  that initially want to take them into use might fail if they get scheduled
+  before the policy is installed. This needs to be figured out and possibly
+  addressed (if the container runtime engine doesn't retry). We might also need
+  a way to ensure that the pod is only run when the policy is installed.
+
+* Currently, the implementation runs a `DaemonSet` that installs the policy in
+  all the nodes in the cluster. We might want to limit this by exposing a
+  `nodeSelector` to the CRD.
+
+* Given that this uses a custom operator and a custom resource, integration
+  with `Pods` and standard kubernetes resources doesn't necessarily use the
+  same names (e.g. why the `usage` section was needed in the first place). In
+  the future, we could consider adding a section to the `seLinuxOptions`
```

Comment:
this would require a change upstream k8s itself, right?  is it better to explore this upstream similar to the seccomp proposed sub-project?

Reply:
This could be an option. However, I decided to post this here first cause it's better to first show something that's working and gain adoption, before trying to submit it there. SELinux is not available for all kube distros after all.